bcipy/tasks/rsvp/stopping_criteria.py

```diff
@@ -0,0 +1,231 @@
+        # see MinIterationsCriteria comment
+        current_seq = len(epoch['list_sti'])
+        if current_seq >= self.max_num_seq:
+            log.debug(
+                "Committing to decision: max iterations have been reached.")
+            return True
+        return False
+
+
+class ProbThresholdCriteria(DecisionCriteria):
+    """Returns true if the commit threshold has been met."""
+
+    def __init__(self, threshold):
+        """ Args:
+                threshold(float \in [0,1]): A threshold on most likely
+                    candidate posterior. If a candidate exceeds a posterior
+                    the system terminates.
+                 """
+        assert 1 >= threshold >= 0, "stopping threshold should be in [0,1]"
+        self.tau = threshold
+
+    def decide(self, epoch):
+        current_distribution = epoch['list_distribution'][-1]
+        if np.max(current_distribution) > self.tau:
+            log.debug("Committing to decision: posterior exceeded threshold.")
+            return True
+        return False
+
+
+class MarginCriteria(DecisionCriteria):
+    """ Stopping criteria based on the difference of two most likely candidates.
+        This condition
+    """
+
+    def __init__(self, margin):
+        """ Args:
+                margin(float \in [0,1]): Minimum distance required between
+                    two most likely competing candidates to trigger termination.
+                    """
+        assert 1 >= margin >= 0, "difference margin should be in [0,1]"
+        self.margin = margin
+
+    def decide(self, epoch):
+        # Get the current posterior probability values
+        p = copy(epoch['list_distribution'][-1])
+        # This criteria compares most likely candidates (best competitors)
+        candidates = [p[idx] for idx in list(np.argsort(p)[-2:])]
+        stopping_rule = np.abs(candidates[0] - candidates[1])
+        d = stopping_rule > self.margin
+        if d:
+            log.debug("Committing to decision: margin is high enough.")
+
+        return d
+
+
+class MomentumCommitCriteria(DecisionCriteria):
+    """ Stopping criteria based on Renyi entropy on the simplex
+        Renyi entropy for infinity and 1 norms are special forms.
+        Attr:
+            alpha(float): a non-negative order for the entropy
+            tau(float): decision threshold
+            """
+
+    def __init__(self, tau, lam):
+        self.lam = lam
+        self.tau = tau
+
+    def reset(self):
+        pass
+
+    def decide(self, epoch):
+        eps = np.power(.1, 6)
+
+        prob_history = copy(epoch['list_distribution'])
+        p = prob_history[-1]
+
+        tmp_p = np.ones(len(p)) * (1 - self.tau) / (len(p) - 1)
+        tmp_p[0] = self.tau
+
+        tmp = -np.sum(p * np.log2(p + eps))
+        tau_ = -np.sum(tmp_p * np.log2(tmp_p + eps))
+
+        tmp_ = np.array(prob_history)
+        mom_ = tmp_[1:] * (
+                np.log(tmp_[1:] + eps) - np.log(tmp_[:-1] + eps))
```

Comment:
The indentation for this line isn't correct. This can all be on one line.

Reply:
I guess by incorrect you mean the coding standards.
All indentation is covered with my on the fly PEP8 checker that has a fixed column length.
Please lemme know of your column length number so I can adjust accordingly.

Reply:
When I ran pylint, I did not come across an indentation error either. 
Side question: Is there a way to adjust the column length?

Reply:
There is a way to change max_line_length. We set that in `setup.cfg`. I'd change it to one line in this case since it doesn't violate line length rules